x-pack/plugins/ml/server/models/data_frame_analytics/analytics_manager.ts

```diff
@@ -527,6 +551,202 @@ export class AnalyticsManager {
     }
   }
 
+  /**
+   * Expanded wrapper of getAnalyticsMap, which also handles generic models that are not tied to an analytics job
+   * Retrieves info about model and ingest pipeline, index, and transforms associated with the model
+   * @param analyticsId
+   * @param modelId
+   */
+  public async extendModelsMap({
+    analyticsId,
+    modelId,
+  }: {
+    analyticsId?: string;
+    modelId?: string;
+  }): Promise<AnalyticsMapReturnType> {
+    const result: AnalyticsMapReturnType = {
+      elements: [],
+      details: {},
+      error: null,
+    };
+    try {
+      if (analyticsId && !modelId) {
+        return this.getAnalyticsMap({
+          analyticsId,
+          modelId,
+        } as GetAnalyticsJobIdArg);
+      }
+
+      await this.initData();
+
+      const modelNodeId = `${modelId}-${JOB_MAP_NODE_TYPES.TRAINED_MODEL}`;
+      const model = modelId ? this.findTrainedModel(modelId) : undefined;
+
+      const isDFAModel = isPopulatedObject(model?.metadata, ['analytics_config']);
+      if (isDFAModel) {
+        return this.getAnalyticsMap({
+          analyticsId,
+          modelId,
+        } as GetAnalyticsModelIdArg);
+      }
+
+      if (modelId && model) {
+        // First, find information about the trained model
+        result.elements.push({
+          data: {
+            id: modelNodeId,
+            label: modelId,
+            type: JOB_MAP_NODE_TYPES.TRAINED_MODEL,
+          },
+        });
+        result.details[modelNodeId] = model;
+
+        // Then, find the pipelines that have the trained model set as index.default_pipelines
+        const pipelinesResponse = await modelsProvider(this._client).getModelsPipelines([modelId]);
+
+        const pipelines = pipelinesResponse.get(modelId);
+
+        if (pipelines) {
+          const pipelineIds = new Set(Object.keys(pipelines));
+          pipelineIds.forEach((pipelineId) => {
+            const pipelineNodeId = `${pipelineId}-${JOB_MAP_NODE_TYPES.INGEST_PIPELINE}`;
+            result.details[pipelineNodeId] = pipelines[pipelineId];
+
+            result.elements.push({
+              data: {
+                id: pipelineNodeId,
+                label: pipelineId,
+                type: JOB_MAP_NODE_TYPES.INGEST_PIPELINE,
+              },
+            });
+
+            result.elements.push({
+              data: {
+                id: `${modelNodeId}~${pipelineNodeId}`,
+                source: modelNodeId,
+                target: pipelineNodeId,
+              },
+            });
+          });
+          const pipelineIdsToDestinationIndices: Record<string, string[]> = {};
+          // // const pipelinesToIndices = new Map();
+          const indicesSettings = await this._client.asCurrentUser.indices.getSettings();
```

Comment:
I've not tested this, but I wonder about using current user here. Getting the settings might be something a viewer user is not allowed to do.

I don't know what level of user will be allowed to use this, it looks like it should be allowed with a `canGetDataFrameAnalytics` permission which is a viewer user.

Also, not a mistake in this PR, but I noticed that the route which uses this code is actually missing its tag. I think it should be `access:ml:canGetDataFrameAnalytics`.
It would be good to add this in this PR if possible.

Reply:
I'm assuming with the new `access:ml:canGetDataFrameAnalytics`, we can use asInternalUser. Otherwise, we can make it so that if user has get indices settings permission, it will show the map. Else, only show partially? [`8f14114` (#162443)](https://github.com/elastic/kibana/pull/162443/commits/8f1411438f8e4e828894aaded973391e9512ef56)